backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/AnnotatingNodeDescriptionProvider.java

```diff
@@ -0,0 +1,170 @@
+import org.eclipse.syson.sysmlcustomnodes.SysMLNoteNodeStyleDescription;
+import org.eclipse.syson.util.AQLUtils;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+import org.eclipse.syson.util.ViewConstants;
+
+/**
+ * Used to create the annotating node description in all diagrams.
+ *
+ * @author jmallet
+ */
+public class AnnotatingNodeDescriptionProvider extends AbstractNodeDescriptionProvider {
+
+    private final EClass eClass;
+
+    private final UtilService utilServices;
+
+    private final IDescriptionNameGenerator descriptionNameGenerator;
+
+    public AnnotatingNodeDescriptionProvider(EClass eClass, IColorProvider colorProvider, IDescriptionNameGenerator nameGenerator) {
+        super(colorProvider);
+        this.eClass = Objects.requireNonNull(eClass);
+        this.descriptionNameGenerator = Objects.requireNonNull(nameGenerator);
+        this.utilServices = new UtilService();
+    }
+
+    /**
+     * Implementers might provide the name of the Usage node description.<br>
+     * Default implementation returns the name based on the name of the semantic type.
+     *
+     * @return the name of the Usage node description
+     */
+    private String getNodeDescriptionName() {
+        return this.getDescriptionNameGenerator().getNodeName(this.eClass);
+    }
+
+    /**
+     * Implementers should provide the expression used to retrieve all semantic candidates.<br>
+     * By default, the expression retrieves all reachable element of the given semantic type.
+     *
+     * @param domainType
+     *            the semantic type of the element.
+     * @return the AQL expression to retrieve all semantic candidates for this node.
+     */
+    private String getSemanticCandidatesExpression(String domainType) {
+        return this.utilServices.getAllReachableExpression(domainType);
+    }
+
+    @Override
+    public NodeDescription create() {
+        String domainType = SysMLMetamodelHelper.buildQualifiedName(this.eClass);
+        return this.diagramBuilderHelper.newNodeDescription()
+                .collapsible(true)
+                .childrenLayoutStrategy(new FreeFormLayoutStrategyDescriptionBuilder().build())
+                .defaultHeightExpression(ViewConstants.DEFAULT_NOTE_HEIGHT)
+                .defaultWidthExpression(ViewConstants.DEFAULT_NOTE_WIDTH)
+                .domainType(domainType)
+                .insideLabel(this.createInsideLabelDescription())
+                .name(this.getNodeDescriptionName())
+                .semanticCandidatesExpression(this.getSemanticCandidatesExpression(domainType))
+                .style(this.createNoteNodeStyle())
+                .userResizable(UserResizableDirection.BOTH)
+                .synchronizationPolicy(SynchronizationPolicy.UNSYNCHRONIZED)
+                .preconditionExpression(AQLUtils.getSelfServiceCallExpression("showAnnotatingNode", List.of("diagramContext", "editingContext")))
```

Comment:
Please use IDiagramContext.DIAGRAM_CONTEXT and IEditingContext.EDITING_CONTEXT instead.